src/managed/Microsoft.Extensions.DependencyModel/RuntimeAssetGroup.cs

```diff
@@ -22,8 +31,35 @@ public RuntimeAssetGroup(string runtime, IEnumerable<string> assetPaths)
         public string Runtime { get; }
 
         /// <summary>
-        /// Gets a list of assets provided in this runtime group
+        /// Gets a list of asset paths provided in this runtime group
+        /// </summary>
+        public IReadOnlyList<string> AssetPaths
+        {
+            get
+            {
+                if (_assetPaths != null)
+                {
+                    return _assetPaths;
+                }
+
+                return _runtimeFiles.Select(file => file.Path).ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Gets a list of RuntimeFiles provided in this runtime group
         /// </summary>
-        public IReadOnlyList<string> AssetPaths { get; }
+        public IReadOnlyList<RuntimeFile> RuntimeFiles
+        {
+            get
+            {
+                if (_runtimeFiles != null)
+                {
+                    return _runtimeFiles;
+                }
+
+                return _assetPaths.Select(path => new RuntimeFile(path, null, null)).ToArray();
```

Comment:
It feels wrong for a property to call `ToArray()` every time it is invoked.

Reply:
Normally this is not hit; it's for compatibility - it has two modes dependency on the ctor invoked. If the "old" ctor is invoked with string but requests RuntimeFile then this would be hit (and vise-versa if the new ctor invoked).